### PR TITLE
add __len__ method to read_root with chunks

### PR DESCRIPTION
**Pull request by pseyfert**
This allows usage in progress bars, such as tqdm
```
for chunk in tqdm(read_root("filename.root", chunksize=100)):
    do_something(chunk)
```

### CODE REVIEW

File: root_pandas/readwrite.py [hunk 1]
```diff
@@ -252,19 +252,23 @@ def read_root(paths, key=None, columns=None, ignore=None, chunksize=None, where=
...
-                    continue
-                if flatten:
-                    arr = do_flatten(arr, flatten)
-                yield convert_to_dataframe(arr, start_index=current_index)
-                current_index += len(arr)
-        return genchunks()
+        class genchunk(object):
+            def __len__(self):
+                return n_chunks
+            def __iter__(self):
```
Comment (chrisburr) [inline root_pandas/readwrite.py]: Can you add a new line before this to conform to PEP8?

File: root_pandas/readwrite.py [hunk 2]
```diff
@@ -252,19 +252,23 @@ def read_root(paths, key=None, columns=None, ignore=None, chunksize=None, where=
...
+                current_index = 0
+                for chunk in range(n_chunks):
+                    arr = root2array(paths, key, all_vars, start=chunk * chunksize, stop=(chunk+1) * chunksize, selection=where, *args, **kwargs)
+                    if len(arr) == 0:
+                        continue
+                    if flatten:
+                        arr = do_flatten(arr, flatten)
+                    yield convert_to_dataframe(arr, start_index=current_index)
+                    current_index += len(arr)
+        return genchunk()
```
Comment (chrisburr) [inline root_pandas/readwrite.py]: And the same before this line

File: root_pandas/readwrite.py (hunk 1 above)
Comment (pseyfert) [reply root_pandas/readwrite.py]: done

File: root_pandas/readwrite.py (hunk 2 above)
Comment (pseyfert) [reply root_pandas/readwrite.py]: done

**Comment by chrisburr** [comment]
Great idea! Thanks 👍

**Comment by eduardo-rodrigues** [comment]
Hi @pseyfert and @chrisburr, may I suggest that you add an example usage to the README file?

**Pull request merged**